File: Tests/HeadElement.Services.Test/fixtures/metaElementsSet.ts
export class MetaElementsSet {

    public metaElements: { [key: string]: HTMLMetaElement };

    public get description() { return this.metaElements["description"]; }
    public get contentType() { return this.metaElements["contentType"]; }
    public get themeColorLight() { return this.metaElements["themeColorLight"]; }
    public get themeColorDark() { return this.metaElements["themeColorDark"]; }
    public get ogType() { return this.metaElements["ogType"]; }

    constructor() {
        this.metaElements = {
            "description": this.createMeta({ name: "description", content: "Hello, World." }),
            "contentType": this.createMeta({ httpEquiv: "content-type", content: "text/html; charset=utf-8" }),
            "themeColorLight": this.createMeta({ name: "theme-color", media: "(prefers-color-scheme: light)", content: "white" }),
            "themeColorDark": this.createMeta({ name: "theme-color", media: "(prefers-color-scheme: dark)", content: "black" }),
            "ogType": this.createMeta({ property: "og:type", content: "website" }),
            "unknownName": this.createMeta({ name: "unknown", content: "unknown content" }),
        };
    }

    public forEach(action: (meta: HTMLMetaElement) => void): void {
        for (const meta of Object.values(this.metaElements)) {
            action(meta);
        }
    }

    private createMeta(props: any): HTMLMetaElement {
        return Object.assign(new HTMLMetaElementMock(), props);
    }
}

class HTMLMetaElementMock {
    public name = '';
    public content = '';
    public httpEquiv = '';
    public media = '';
    public getAttribute(name: string): string | null {
        return (this as any)[name] || null;
    }
}

File: Tests/HeadElement.ServerPrerendering.Test/Fixtures/MetaElementsSet.cs
using AngleSharp.Html.Dom;
using AngleSharp.Html.Parser;

namespace HeadElement.ServerPrerendering.Test.Fixtures;

public class MetaElementsSet
{
    public IReadOnlyDictionary<string, IHtmlMetaElement> MetaElements { get; }

    public IHtmlMetaElement Description => this.MetaElements[nameof(this.Description)];

    public IHtmlMetaElement ContentType => this.MetaElements[nameof(this.ContentType)];

    public IHtmlMetaElement ThemeColorLight => this.MetaElements[nameof(this.ThemeColorLight)];

    public IHtmlMetaElement ThemeColorDark => this.MetaElements[nameof(this.ThemeColorDark)];

    public IHtmlMetaElement OgType => this.MetaElements[nameof(this.OgType)];

    public IHtmlMetaElement UnknownName => this.MetaElements[nameof(this.UnknownName)];


    public MetaElementsSet()
    {
        this.MetaElements = new Dictionary<string, IHtmlMetaElement>
        {
            [nameof(this.Description)] = CreateMetaElement(@"<meta name=""description"" content=""Hello, World."" />"),
            [nameof(this.ContentType)] = CreateMetaElement(@"<meta http-equiv=""content-type"" content=""text/html; charset=utf-8"" />"),
            [nameof(this.ThemeColorLight)] = CreateMetaElement(@"<meta name=""theme-color"" media=""(prefers-color-scheme: light)"" content=""white""/>"),
            [nameof(this.ThemeColorDark)] = CreateMetaElement(@"<meta name=""theme-color"" media=""(prefers-color-scheme: dark)"" content=""black""/>"),
            [nameof(this.OgType)] = CreateMetaElement(@"<meta property=""og:type"" content=""website""/>"),
            [nameof(this.UnknownName)] = CreateMetaElement(@"<meta name=""unknown"" content=""unknown content""/>"),
        };
    }

    public void ForEach(Action<IHtmlMetaElement> action)
    {
        foreach (var metaElement in this.MetaElements.Values)
        {
            action.Invoke(metaElement);
        }
    }

    private static IHtmlMetaElement CreateMetaElement(string content)
    {
        var parser = new HtmlParser();
        using var doc = parser.ParseDocument(
            "<html><head>" +
            content +
            "</head></html>");
        var meta = doc.QuerySelector("meta") as IHtmlMetaElement;
        if (meta == null) throw new Exception("A meta element was not found.");
        return meta;
    }
}
